### PR TITLE
Concat now working with new architecture

### PR DESCRIPTION
**Pull request by osalpekar**
Also added some test cases involving `pandas.Series` objects.

### CODE REVIEW

File: modin/pandas/concat.py [hunk 1]
```diff
@@ -65,89 +65,16 @@ def concat(objs,
...
-                    all_index[i], final_index, axis,
-                    len(objs[0]._block_partitions.T)
-                ] + part.tolist()),
-                num_return_vals=len(objs[0]._block_partitions.T))
-            for i in range(len(objs)) for part in objs[i]._block_partitions.T
-        ]).T
-
-    return DataFrame(
-        block_partitions=new_blocks, columns=final_columns, index=final_index)
+    objs = [DataFrame(obj) if not isinstance(obj, DataFrame) else
```
Comment (devin-petersohn) [inline modin/pandas/concat.py]: Does this present a `pandas.Series` in the correct format? The name of the `Series` should become the name of the column, I believe.

Comment (osalpekar) [reply modin/pandas/concat.py]: As it currently is, it handles every case correctly except concatenating a `pandas.DataFrame` to `pandas.Series` on `axis=0`, in which case pandas actually ignores the name of the Series. I've updated the PR to mirror this.

File: modin/data_management/data_manager.py [hunk 2]
```diff
@@ -233,9 +233,9 @@ def _join_data_manager(self, other, **kwargs):
 
         # This stage is to efficiently get the resulting columns, including the
         # suffixes.
-        self_proxy = pandas.DataFrame(columns=self.columns)
-        other_proxy = pandas.DataFrame(columns=other.columns)
-        new_columns = self_proxy.join(other_proxy, lsuffix=lsuffix, rsuffix=rsuffix).columns
+        self_proxy_columns = pandas.DataFrame(columns=self.columns).columns
+        other_proxy_columns = pandas.DataFrame(columns=other.columns).columns
+        new_columns = self_proxy_columns.append(others_proxy_columns)
```
Comment (devin-petersohn) [inline modin/data_management/data_manager.py]: I might have missed this before. This won't end with the correct column names if you don't join. The point here is to get the order of the columns correct and add the correct suffixes to each.

File: modin/data_management/data_manager.py [hunk 3]
```diff
@@ -267,9 +265,9 @@ def _join_list_of_managers(self, others, **kwargs):
 
         # This stage is to efficiently get the resulting columns, including the
         # suffixes.
-        self_proxy = pandas.DataFrame(columns=self.columns)
-        others_proxy = [pandas.DataFrame(columns=other.columns) for other in others]
-        new_columns = self_proxy.join(others_proxy, lsuffix=lsuffix, rsuffix=rsuffix).columns
+        self_proxy_columns = pandas.DataFrame(columns=self.columns).columns
+        others_proxy_columns = [pandas.DataFrame(columns=other.columns).columns for other in others]
+        new_columns = self_proxy_columns.append(others_proxy_columns)
```
Comment (devin-petersohn) [inline modin/data_management/data_manager.py]: Same as above.

**Pull request merged**